File: lib/src/models/order_item.dart
import 'package:sneakerx/src/models/product.dart';
import 'package:sneakerx/src/models/product_sku.dart';

class OrderItem {
  final int orderItemId;
  final int shopOrderId;
  final Product product;
  final ProductSku sku;
  final String productNameSnapshot;
  final String skuNameSnapshot;
  final double priceAtPurchase;
  final int quantity;

  OrderItem({
    required this.orderItemId,
    required this.shopOrderId,
    required this.product,
    required this.sku,
    required this.productNameSnapshot,
    required this.skuNameSnapshot,
    required this.priceAtPurchase,
    required this.quantity,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      orderItemId: json['orderItemId'],
      shopOrderId: json['shopOrderId'],
      product: Product.fromJson(json['product'] as Map<String, dynamic>),
      sku: ProductSku.fromJson(json['sku'] as Map<String, dynamic>),
      productNameSnapshot: json['productNameSnapshot'],
      skuNameSnapshot: json['skuNameSnapshot'],
      priceAtPurchase: json['priceAtPurchase'],
      quantity: json['quantity']
    );
  }
}

File: src/main/java/com/example/sneakerx/entities/OrderItem.java
package com.example.sneakerx.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "order_items")
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer orderItemId;

    @ManyToOne
    @JoinColumn(name = "shop_order_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private ShopOrder shopOrder;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Product product;

    @ManyToOne
    @JoinColumn(name = "sku_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private ProductSku sku;

    private String productNameSnapshot;
    private String skuNameSnapshot;
    private Double priceAtPurchase;
    private Integer quantity;
}
